### ISSUE TITLE
`/usr/bin/python3: Error while finding module specification for 'synapse.app.homeserver' (ModuleNotFoundError: No module named 'synapse.app')`

### ISSUE THREAD

**Issue by HammyHavoc** [matrix-org/synapse#14717]
### Description

Updated to the latest version (1.74), Synapse won't start, and getting this error:

`/usr/bin/python3: Error while finding module specification for 'synapse.app.homeserver' (ModuleNotFoundError: No module named 'synapse.app')`

### Steps to reproduce

- start Synapse

### Homeserver

matrix.splitanatom.com

### Synapse Version

1.74

### Installation Method

Docker (matrixdotorg/synapse)

### Database

postgresql; single server; not ported; not previously restored from a backup

### Workers

Single process

### Platform

Docker container on Unraid

### Relevant log output

```shell
`/usr/bin/python3: Error while finding module specification for 'synapse.app.homeserver' (ModuleNotFoundError: No module named 'synapse.app')`
```

**Comment by moirk** [comment]
Hello, same issue here Unraid Also.

**Comment by HammyHavoc** [comment]
My temporary fix is changing the repo in the config for the container to
the previous version number just to get it back up and running. Hope that
gets you back up in the meantime! :- )

**Comment by moirk** [comment]
Unfortunatly tried this and have another error :
`Error during startup:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/twisted/internet/tcp.py", line 1334, in startListening
    skt.bind(addr)
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/synapse/app/_base.py", line 240, in wrapper
    await cb(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/synapse/app/homeserver.py", line 386, in start
    await _base.start(hs)
  File "/usr/local/lib/python3.10/dist-packages/synapse/app/_base.py", line 500, in start
    hs.start_listening()
  File "/usr/local/lib/python3.10/dist-packages/synapse/app/homeserver.py", line 283, in start_listening
    self._listener_http(self.config, listener)
  File "/usr/local/lib/python3.10/dist-packages/synapse/app/homeserver.py", line 166, in _listener_http
    ports = listen_tcp(
  File "/usr/local/lib/python3.10/dist-packages/synapse/app/_base.py", line 353, in listen_tcp
    check_bind_error(e, address, bind_addresses)
  File "/usr/local/lib/python3.10/dist-packages/synapse/app/__init__.py", line 53, in check_bind_error
    raise e
  File "/usr/local/lib/python3.10/dist-packages/synapse/app/_base.py", line 351, in listen_tcp
    r.append(reactor.listenTCP(port, factory, backlog, address))
  File "/usr/local/lib/python3.10/dist-packages/twisted/internet/posixbase.py", line 565, in listenTCP
    p.startListening()
  File "/usr/local/lib/python3.10/dist-packages/twisted/internet/tcp.py", line 1336, in startListening
    raise CannotListenError(self.interface, self.port, le)
twisted.internet.error.CannotListenError: Couldn't listen on [:8008: [Errno -2] Name or service not known.`

**Comment by moirk** [comment]
I managed to make the last version work.
I changed my bind_addresses to : 
`bind_addresses: ['0.0.0.0']`
was  `bind_addresses: '[0.0.0.0]'` before

**Comment by HammyHavoc** [comment]
> I managed to make the last version work. I changed my bind_addresses to : `bind_addresses: ['0.0.0.0']` was `bind_addresses: '[0.0.0.0]'` before

The last version being the most recent, or the version prior that previously worked?

**Comment by moirk** [comment]
sorry was not clear, I managed to make the most recent version available to work (1.74.0) with the modification stated above.

**Comment by Nelinski** [comment]
I've got this issue too, above suggestion doesn't help unfortunately.

**Comment by Nelinski** [comment]
I had this issue using Avhost's docker on Unraid. They've fixed it now and rebuilt 1.74:

https://hub.docker.com/layers/avhost/docker-matrix/v1.74.0/images/sha256-66ea506afc516a234bc8d66e2307403801bf732906ba32393d1f9eb16f69fe6e?context=explore

Not sure if you're using similar but all working for me now after completely deleting the image and reinstalling (check you're using the same hash).

**Comment by clokep** [comment]
This sounds like an issue with an unofficial distribution -- I'm going to close this issue. Please report bugs on the Unraid issue tracker.

----

If anyone is seeing this with the *official* Docker image (from https://hub.docker.com/r/matrixdotorg/synapse) then please shout and we can reopen.
